sklearn/decomposition/_dict_learning.py

```diff
@@ -2381,9 +2382,10 @@ def fit(self, X, y=None):
             for i, batch in zip(range(n_steps), batches):
                 X_batch = X_train[batch]
 
-                batch_cost = self._minibatch_step(
-                    X_batch, dictionary, self._random_state, i
-                )
+                with config_context(assume_finite=True, skip_parameter_validation=True):
+                    batch_cost = self._minibatch_step(
+                        X_batch, dictionary, self._random_state, i
+                    )
```

Comment:
I think I am still in favor of merging #25490 first.

But then I am not opposed to merge this one as well, but not for the `_minibatch_step` call anymore if #25490 is merged.

Reply:
#25490 only gets rid of 1 layer of validation (there are 4 validations in total) so even if it gets merged, we'll still need the context manager around the `_minibatch_step`.